goas/cl/ledger/src/partial_tx.rs

```diff
@@ -1,76 +1,120 @@
 use std::collections::BTreeMap;
 
-use ledger_proof_statements::death_constraint::DeathConstraintPublic;
+use ledger_proof_statements::{
+    death_constraint::DeathConstraintPublic,
+    ptx::{PtxPrivate, PtxPublic},
+};
 
 use crate::{
-    death_constraint::DeathProof, error::Result, input::ProvedInput, output::ProvedOutput,
+    death_constraint::DeathProof, error::{Error, Result}
 };
 
-#[derive(Debug, Clone)]
-pub struct PartialTxInput {
-    pub input: ProvedInput,
-    pub death: DeathProof,
-}
+const MAX_NOTE_COMMS: usize = 2usize.pow(8);
 
-impl PartialTxInput {
-    fn verify(&self, ptx_root: cl::PtxRoot) -> bool {
-        let nf = self.input.input.input.nullifier;
-        self.input.input.input.death_cm == self.death.death_commitment() // ensure the death proof is actually for this input
-            && self.input.verify() // ensure the input proof is valid
-            && self.death.verify(DeathConstraintPublic { nf, ptx_root }) // verify the death constraint was satisfied
-    }
-}
 
 pub struct ProvedPartialTx {
-    pub inputs: Vec<PartialTxInput>,
-    pub outputs: Vec<ProvedOutput>,
+    pub ptx: cl::PartialTx,
+    pub cm_root: [u8;32],
+    pub death_proofs: BTreeMap<cl::Nullifier, DeathProof>,
+    pub risc0_receipt: risc0_zkvm::Receipt,
```

Comment:
should we remove explicit mentions to risc0 here? Not something to do in this PR as It was the case before as well, just thinking out loud